File: framework/languages/dotnet/e2e/PubSub/Shared/Messages.cs
namespace PubSub.Shared;

public static class PubSubNames
{
    public const string Channel = "events";
    public const string SubscriberSocketSource = Channel + ".subscriber";
    public const string PublisherSocketSource = Channel + ".publisher";
    public const string MainTopic = "orders";
    public const string OtherTopic = "billing";
}

public sealed record EventMsg(string RunId, int Sequence, string Value);

public sealed record MissingEventMsg(string RunId, int Sequence, string Value);

public sealed record DrainResultRes(string Result, string? Reason = null);

public sealed record EvidenceWaitReq(
    string[] ContainsAll,
    string[][] ContainsAnyGroups,
    int TimeoutMilliseconds = 10000)
{
    public int AfterIndex { get; init; }

    public string[][] ContainsAllLineGroups { get; init; } = [];
    public string[][] ContainsAnyLineGroups { get; init; } = [];
}

File: framework/languages/node/e2e/PubSub/Shared/messages.ts
export const PubSubNames = {
  channel: 'events',
  mainTopic: 'orders',
  otherTopic: 'billing'
} as const;

export const PacketNames = {
  eventMsg: 'EventMsg',
  missingEventMsg: 'MissingEventMsg'
} as const;

export class EventMsg {
  constructor(readonly runId: string, readonly sequence: number, readonly value: string) {}
}

export class MissingEventMsg {
  constructor(readonly runId: string, readonly sequence: number, readonly value: string) {}
}

export interface EvidenceWaitReq {
  readonly containsAll?: readonly string[];
  readonly containsAnyGroups?: readonly (readonly string[])[];
  readonly containsAllLineGroups?: readonly (readonly string[])[];
  readonly containsAnyLineGroups?: readonly (readonly string[])[];
  readonly timeoutMilliseconds?: number;
  readonly afterIndex?: number;
}
